src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CsvDataExtractionService.kt

```diff
@@ -13,25 +14,37 @@ import java.time.LocalDateTime
 import java.time.temporal.ChronoUnit.DAYS
 import java.util.stream.Stream
 import kotlin.streams.asSequence
+import kotlin.system.measureTimeMillis
 
 @Service
 class CsvDataExtractionService(
   private val videoBookingEventRepository: VideoBookingEventRepository,
   private val locationsService: LocationsService,
 ) {
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
 
   @Transactional(readOnly = true)
   fun courtBookingsByHearingDateToCsv(fromDate: LocalDate, toDate: LocalDate, csvOutputStream: OutputStream) {
     checkDaysBetweenDoesNotExceedAYear(fromDate, toDate)
 
-    writeCourtBookingsToCsv(videoBookingEventRepository.findByMainDateBetween(true, fromDate, toDate), csvOutputStream)
+    val elapsed = measureTimeMillis {
+      writeCourtBookingsToCsv(videoBookingEventRepository.findByMainDateBetween(true, fromDate, toDate), csvOutputStream)
+    }
+
+    log.info("CSV: time taken downloading court bookings by hearing date from $fromDate to $toDate in millis=$elapsed")
```

Comment:
You can see a visible time difference in the logs after locations have been cached.